src/linux/component.rs

```diff
@@ -176,14 +405,12 @@ pub(crate) fn get_components() -> Vec<Component> {
         }
         components.sort_by(|c1, c2| c1.label.to_lowercase().cmp(&c2.label.to_lowercase()));
     }
-    if is_file("/sys/class/thermal/thermal_zone0/temp") {
-        // Specfic to raspberry pi.
-        components.push(Component::new(
-            "CPU".to_owned(),
-            Path::new("/sys/class/thermal/thermal_zone0/temp").to_path_buf(),
-            None,
-            None,
-        ));
+
+    // Specfic to raspberry pi tries to avoid it, better info from hwmon.
+    if is_file("/sys/class/thermal/thermal_zone0/temp")
+        && components.iter().position(|c| c.label == "CPU").is_none()
+    {
+        components.push(Component::new_raspberry_pi(String::from("CPU")));
     }
```

Comment:
May be better to feature gate it for rasp pi?

Reply:
Wrong name, would rather say "compile it only on rpy:
https://www.reddit.com/r/rust/comments/9yve3r/how_to_detect_if_im_on_a_raspberry_pi/

```
 #[cfg(all(target_arch = "arm", target_os="linux", target_env="gnu"))]
 ```
 No idea, may need a test some of the hack for the py might be not needed